src/logmessage.cpp

```diff
@@ -122,7 +122,11 @@ namespace g3 {
 
 
    std::string LogMessage::timestamp(const std::string& time_look) const {
-      return  localtime_formatted(_timestamp, time_look);
+      std::ostringstream stringStream;
+
+      stringStream << _timestamp.tv_nsec;
+
+      return g3::localtime_formatted(_timestamp.tv_sec, (time_look.empty() ? g3::internal::date_formatted + " " + g3::internal::time_formatted : time_look)) + "|" + stringStream.str() + "|";
```

Comment:
I tried to avoid touching formatting part including localtime_formatted function before we sort out data structure and generation step. localtime_formatted accepts only seconds part so I appended fractional part between pipes just to print it out to sinks.
Now data structure and generation are sorted so I will clean up formatting part ;)